packages/compiler/lib/decorators.ts

```diff
@@ -813,3 +813,62 @@ export function isDeprecated(program: Program, type: Type): boolean {
 export function getDeprecated(program: Program, type: Type): string | undefined {
   return program.stateMap(deprecatedKey).get(type);
 }
+
+const overloadedByKey = Symbol("overloadedByKey");
+const overloadsOperationKey = Symbol("overloadsOperation");
+
+/**
+ * `@overload` - Indicate that the target overloads (specializes) the overloads type.
+ * @param context DecoratorContext
+ * @param target The specializing operation declaration
+ * @param overloads The operation to be overloaded.
+ */
+export function $overload(
+  context: DecoratorContext,
+  target: OperationType,
+  overloads: OperationType
+) {
+  // TODO: Verify that the target is a subtype of the overloads instance (e.g. doesn't expand on the set
+  // of parameters or return types for the overloads function)
+  if (!validateDecoratorTarget(context, target, "@overload", "Operation")) {
+    return;
+  }
+  context.program.stateMap(overloadsOperationKey).set(target, overloads);
+  if (overloads.kind != "Operation") {
+    context.program.reportDiagnostic({
+      code: "cadl-overload-wrong-type",
+      severity: "error",
+      message: "You can only @overload other operations.",
+      target: target,
+    });
+  }
+  const existingOverloads =
+    getOverloadedBy(context.program, overloads) || new Array<OperationType>();
+  context.program.stateMap(overloadedByKey).set(overloads, existingOverloads.concat(target));
+}
+
+/**
+ * Get all operations that are marked as overloads of the given operatoin
+ * @param context
+ * @param operation
+ * @returns
+ */
+export function getOverloadedBy(
```

Comment:
Might call this `getOverloads`

Reply:
getOverloadedBy [executed against the overloaded method]
getOverloads

isOverloading [executed against the overloading method]
isOverload
getOverloadTarget

Reply:
Went with `getOverloads` and `getOverloadedOperation`.